src/tools/tag.ts

```diff
@@ -0,0 +1,1080 @@
+        wiqlConditions.push(`[System.AreaPath] IN ('${areaPath.join("','")}')`);
+      }
+      if (workItemType && workItemType.length > 0) {
+        wiqlConditions.push(`[System.WorkItemType] IN ('${workItemType.join("','")}')`);
+      }
+      if (state && state.length > 0) {
+        wiqlConditions.push(`[System.State] IN ('${state.join("','")}')`);
+      }
+      if (assignedTo && assignedTo.length > 0) {
+        wiqlConditions.push(`[System.AssignedTo] IN ('${assignedTo.join("','")}')`);
+      }
+
+      tag.forEach((t) => {
+        wiqlConditions.push(`[System.Tags] CONTAINS '${t}'`);
+      });
+
+      const wiqlQuery = `
+        SELECT [System.Id], [System.Title], [System.State], [System.Tags]
+        FROM WorkItems
+        WHERE ${wiqlConditions.join(" AND ")}
+        ORDER BY [System.ChangedDate] DESC
+      `;
+
+      const wiqlUrl = `https://dev.azure.com/${orgName}/${project?.[0] || ""}/_apis/wit/wiql?api-version=7.1-preview.2`;
+
+      const wiqlResponse = await fetch(wiqlUrl, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "Authorization": `Bearer ${accessToken.token}`,
+          "User-Agent": userAgentProvider(),
+        },
+        body: JSON.stringify({ query: wiqlQuery }),
+      });
+
+      if (!wiqlResponse.ok) {
+        throw new Error(`WIQL API error: ${wiqlResponse.status} ${wiqlResponse.statusText}`);
+      }
+
+      const wiqlResult = await wiqlResponse.json();
+      return {
+        content: [{ type: "text", text: JSON.stringify(wiqlResult.workItems, null, 2) }],
+      };
+    }
+  );
+
+  server.tool(
+    TAG_TOOLS.list_project_workitem_tags,
+    "List all tags for a repository.",
+    {
+      project: z.string().describe("Project name or ID"),
+    },
+    async ({ project }) => {
+      try {
+        const accessToken = await tokenProvider();
+        const url = `https://dev.azure.com/${orgName}/${project}/_apis/wit/tags/`;
+        const response = await fetch(url, {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+            "Authorization": `Bearer ${accessToken.token}`,
+            "User-Agent": userAgentProvider(),
+          },
+        });
+        const tags = await response.json();
+        return {
+          content: [
+            {
+              type: "text",
+              text: JSON.stringify(
+                tags.value.map((tag: { name: string }) => tag.name),
+                null,
+                2
+              ),
+            },
+          ],
+        };
+      } catch (error) {
+        return {
+          content: [
+            {
+              type: "text",
+              text: `Error fetching tags: ${error instanceof Error ? error.message : String(error)}`,
+            },
+          ],
+          isError: true,
+        };
+      }
+    }
+  );
+
+  server.tool(
+    TAG_TOOLS.workitem_tags_comprehensive_analytics,
```

Comment:
We can't implement this as is. The volume of WIQL statements run is not going to scale properly for organizations that have tens of thousands of tags. We will need to do something else or put this on hold